src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCallerGenotypingEngine.java

```diff
@@ -556,19 +557,11 @@ private VariantContext addNonRefSymbolicAllele(final VariantContext mergedVC) {
                                                                       final byte[] ref,
                                                                       final SimpleInterval refLoc,
                                                                       final List<VariantContext> activeAllelesToGenotype) {
-        final boolean in_GGA_mode = !activeAllelesToGenotype.isEmpty();
+        //TODO: maintaining previous behavior, which cleared the TreeSet from buildEventMaps before adding only the GGA positions
+        //TODO: This seems wrong -- GGA mode shouldn't discard non-GGA loci, right?
```

Comment:
I had always assumed that GGA mode is supposed to discover alleles as usual but *in addition* force specified alleles to be considered.  However, I recently saw some other code that considers *only* the GGA alleles, as was your understanding.

As for how GGA mode works, take a look at `HaplotypeCallerEngine::callRegion`.  Near the top of the method you will find the following code to grab the GGA alleles for a particular assembly region:

```java
for ( final VariantContext vc : features.getValues(hcArgs.alleles) ) {
    if ( vc.isNotFiltered() ) {
        givenAlleles.add(vc); // do something with these VCs during GGA mode
    }
}
```
The key piece of code here is `features.getValues(hcArgs.alleles)`, where `features` is a `FeatureContext`, that is, an interval of an input vcf corresponding to the assembly region, and `hcArgs.alleles` is a `FeatureInput<VariantContext>`, that is, a vcf.  Thus we can see that on the command line one provides a vcf full of GGA alleles, as opposed to an explicit list of alleles, and that `HaplotypeCallerEngine` queries this vcf for alleles overlapping each assembly region.

PS I didn't know how this worked until you asked this question.  GGA mode is definitely not common knowledge.
`